keras/callbacks.py

```diff
@@ -985,23 +985,73 @@ def on_epoch_end(self, epoch, logs=None):
 
 @keras_export("keras.callbacks.TerminateOnNaN")
 class TerminateOnNaN(Callback):
-    """Callback that terminates training when a NaN loss is encountered."""
+    """Callback that terminates training when a NaN loss is encountered.
 
-    def __init__(self):
+    Args:
+        check_freq: `'batch'` or `'epoch'` or integer. When using `'epoch'`,
+            checks the losses for NaN after every epoch. If using an integer, 
+            let's say `1000`, losses will be checked every 1000 batches.
+            `'batch'` is a synonym for `1`, meaning that check will happen 
+            every batch. Defaults to `'batch'`.
+    """
+
+    def __init__(self, check_freq="batch"):
         super().__init__()
+        self.check_freq = 1 if check_freq == "batch" else check_freq
+        if self.check_freq != "epoch" and not isinstance(self.check_freq, int):
+            raise ValueError(
+                "Unrecognized check_freq: {}".format(self.check_freq)
```

Comment:
It is helpful to additionally mention what was expected.
Also, f-strings are a bit more elegant. How about:
```python
f"Unrecognized check_freq: {self.save_freq}"
"Expected 'batch', 'epoch' or integer"
```

Reply:
updated